checks/evaluation/pinned_dependencies.go

```diff
@@ -90,10 +84,10 @@ func PinningDependencies(name string, c *checker.CheckRequest,
 				Snippet:     rr.Location.Snippet,
 				Remediation: generateRemediation(remediationMetadata, &rr),
 			})
-
-			// Update the pinning status.
-			updatePinningResults(&rr, &wp, pr)
 		}
+
+		// Update the pinning status.
+		updatePinningResults(&rr, &wp, pr)
```

Comment:
so we previously didn't update results when rr.Msg != nil, which was used as an indicator of a parsing error:
https://github.com/ossf/scorecard/blob/08e7fb89f49f8d65edacca3190283e43ba566fb6/checks/raw/pinned_dependencies.go#L395-L400

Is this intentional moving this out?

Reply:
Definetely not intentional. The behavior we want is "if dependency has a parsing error, log error and don't count it on score", is that it?

Reply:
I'll add some validation in the evaluation for when a dependency does not have a `Pinned` state, like in the case of a parsing error, so we have a decent error message. And also make sure we don't count dependencies when we already know they had parsing problems. Otherwise it will break the code in runtime, thanks Spencer!

Reply:
> Definitely not intentional. The behavior we want is "if dependency has a parsing error, log error and don't count it on score", is that it?

This is what the existing code did, but only in this one spot